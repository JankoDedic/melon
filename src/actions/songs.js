import uuid from 'uuid';

export const addSong = (song) => {
  const { title = '', artists = '' } = song;
  return {
    type: 'ADD_SONG',
    song: { id: uuid.v4(), title, artists },
  };
};

export const editSong = (id, updates) => ({
  type: 'EDIT_SONG',
  id,
  updates,
});
