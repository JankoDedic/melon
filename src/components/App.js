import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import Player from './Player';

export const App = ({ isUserAuthenticated }) => (
  <div>
    <AppRouter />
    {isUserAuthenticated && <Player />}
  </div>
);

const mapStateToProps = (state) => ({
  isUserAuthenticated: !!state.auth.uid,
});

export default connect(mapStateToProps)(App);
