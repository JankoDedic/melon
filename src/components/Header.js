import React from 'react';

export default () => (
  <div className="header-container">
    <div className="header">
      <div className="header__logo">
        Melon
      </div>
      <button className="header__logout">Logout</button>
    </div>
  </div>
);
