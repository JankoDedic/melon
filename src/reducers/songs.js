const defaultSongs = [
  { title: 'This Is What You Came For', artists: 'Calvin Harris, Rihanna' },
  { title: 'Your Way', artists: 'Frank Sinatra' }
];

export default (songs = defaultSongs, action) => {
  return songs;
};

