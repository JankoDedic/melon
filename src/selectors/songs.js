export default (songs, filters) => {
  return songs.filter((song) => {
    const doesTitleMatch = song.title.toLowerCase().includes(filters.text.toLowerCase());
    const doArtistsMatch = song.artists.toLowerCase().includes(filters.text.toLowerCase());
    return doesTitleMatch || doArtistsMatch;
  });
};
