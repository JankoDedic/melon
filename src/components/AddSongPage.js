import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startAddSong } from '../actions/songs';

class AddSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startAddSong(song);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <SongForm formTitle="Add Song" onSubmit={this.handleSubmit} />
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddSong: (song) => dispatch(startAddSong(song)),
});

export default connect(undefined, mapDispatchToProps)(AddSongPage);
