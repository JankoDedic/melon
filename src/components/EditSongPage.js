import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startEditSong, removeSong } from '../actions/songs';

class EditSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startEditSong(this.props.song.id, song);
    this.props.history.push('/dashboard');
  }
  handleRemove = () => {
    this.props.removeSong(this.props.song.id);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <div>
        <SongForm
          formTitle="Edit Song"
          onSubmit={this.handleSubmit}
          song={this.props.song}
        />
        <button
          className="form-button"
          onClick={this.handleRemove}
        >
          Remove Song
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  song: state.songs.find((song) => song.id === props.match.params.id),
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  startEditSong: (id, updates) => dispatch(startEditSong(id, updates)),
  removeSong: (id) => dispatch(removeSong(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditSongPage);
