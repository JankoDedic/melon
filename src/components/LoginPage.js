import React from 'react';
import { connect } from 'react-redux';

import Logo from './Logo';
import { startLogin, startAnonymousLogin } from '../actions/auth';

export const LoginPage = ({ startLogin, startAnonymousLogin }) => (
  <div className="login-page-container">
    <div className="login-page">
      <Logo className="login-page__logo" />
      <div className="login-page__slogan">
        One music collection. Any stream.
      </div>
      <button
        type="button"
        className="login-page__button"
        onClick={startLogin}
      >
      </button>
      <button
        type="button"
        className="login-page__guest-login-button"
        onClick={startAnonymousLogin}
      >
        Login as Guest
      </button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startAnonymousLogin: () => dispatch(startAnonymousLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
