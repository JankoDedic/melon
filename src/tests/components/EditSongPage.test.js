import React from 'react';
import { shallow } from 'enzyme';

import { EditSongPage } from '../../components/EditSongPage';
import songs from '../fixtures/songs';

let history, song, startEditSong, startRemoveSong, wrapper;

beforeEach(() => {
  history = { push: jest.fn() };
  song = songs[0];
  startEditSong = jest.fn();
  startRemoveSong = jest.fn();
  wrapper = shallow(
    <EditSongPage
      history={history}
      song={song}
      startEditSong={startEditSong}
      startRemoveSong={startRemoveSong}
    />
  );
});

test('EditSongPage is rendered properly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('EditSongPage handles form submission', () => {
  wrapper.find('SongForm').prop('onSubmit')(song);
  expect(startEditSong).toHaveBeenLastCalledWith(song.id, song);
  expect(history.push).toHaveBeenLastCalledWith('/dashboard');
});

test('EditSongPage handles song removal', () => {
  wrapper.find('SongForm').prop('onRemove')(song.id);
  expect(startRemoveSong).toHaveBeenLastCalledWith(song.id);
  expect(history.push).toHaveBeenLastCalledWith('/dashboard');
});
