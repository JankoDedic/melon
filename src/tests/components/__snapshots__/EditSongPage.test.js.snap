// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditSongPage is rendered properly 1`] = `
<div>
  <SongForm
    formTitle="Edit Song"
    onRemove={[Function]}
    onSubmit={[Function]}
    removeButton={true}
    song={
      Object {
        "artists": "Muse",
        "id": "1",
        "title": "Feeling Good",
        "url": "https://www.youtube.com/watch?v=CmwRQqJsegw",
      }
    }
  />
</div>
`;
